.github/workflows/ci-workflow.yml

```diff
@@ -10,75 +10,20 @@ on:
 
 jobs:
   build_application:
-    strategy:
-      matrix:
-        sdk:
-        - path: $NANOS_SDK
-          name: nanos
-        - path: $NANOX_SDK
-          name: nanox
-        - path: $NANOSP_SDK
-          name: nanosp
-
-    runs-on: ubuntu-latest
-
-    container:
-      image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder:latest
-
-    steps:
-      - name: Clone
-        uses: actions/checkout@v3
-
-      - name: Build application
-        run: |
-          make BOLOS_SDK=${{ matrix.sdk.path }} DEBUG=1
-          cp bin/app.elf bin/solana_${{ matrix.sdk.name }}.elf
-
-      - name: Upload application binaries
-        uses: actions/upload-artifact@v3
-        with:
-          name: solana_binaries
-          path: ./bin/solana_${{ matrix.sdk.name }}.elf
-          if-no-files-found: error
-
-  job_scan_build:
-    name: Clang Static Analyzer
-    needs: build_application
-    runs-on: ubuntu-latest
-    container:
-      image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder:latest
-    steps:
-      - uses: actions/checkout@v3
-      - name: Build with Clang Static Analyzer
-        run: |
-          make clean
-          scan-build --use-cc=clang -analyze-headers -enable-checker security -enable-checker unix -enable-checker valist -o scan-build --status-bugs make default
-      - uses: actions/upload-artifact@v3
-        if: failure()
-        with:
-          name: scan-build
-          path: scan-build
-
-  python_tests_nano:
-    name: Ragger tests
+    name: Build application using the reusable workflow
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_build.yml@v1
+    with:
+      upload_app_binaries_artifact: compiled_app_binaries
+      run_for_devices: '["nanos", "nanox", "nanosp"]'
+
+  ragger_tests:
+    name: Run ragger tests using the reusable workflow
     needs: build_application
-    runs-on: ubuntu-latest
-    steps:
-      - name: Clone
-        uses: actions/checkout@v3
-      - name: Download all binaries
-        uses: actions/download-artifact@v3
-      - name: Gather artifacts
-        run: |
-          mkdir tests/elfs
-          mv *_binaries/*.elf tests/elfs
-      - name: Install APT dependencies
-        run: sudo apt-get update && sudo apt-get install -y qemu-user-static
-      - name: Install Ragger
-        run: |
-          pip install --extra-index-url https://test.pypi.org/simple/ -r tests/python/requirements.txt
-      - name: Run tests
-        run: pytest tests/python/ -v --tb=short --device all
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_ragger_tests.yml@v1
```

Comment:
Don't you also want to add the rule enforcer job ?

Reply:
I'll do it when it's all merged, too many branches on top of branches